src/refactoring/components/Cart/SelectCoupon.tsx

```diff
@@ -0,0 +1,36 @@
+import { Coupon } from "../../../types";
+
+interface SelectCouponProps {
+  coupons: Coupon[];
+  selectedCoupon: Coupon | null;
+  applyCoupon: (coupon: Coupon) => void;
+}
+
+export const SelectCoupon = ({ 
+  coupons, 
+  selectedCoupon, 
+  applyCoupon 
+}: SelectCouponProps) => {
+  return (
+    <div className="mt-6 bg-white p-4 rounded shadow">
+      <h2 className="text-2xl font-semibold mb-2">쿠폰 적용</h2>
+      <select
+        onChange={(e) => applyCoupon(coupons[parseInt(e.target.value)])}
+        className="w-full p-2 border rounded mb-2"
+      >
```

Comment:
```suggestion
export const SelectCoupon = ({ 
  coupons, 
  selectedCoupon, 
  applyCoupon 
}: SelectCouponProps) => {
  const applyCouponHandler = (e : React.ChangeEvent<HTMLElement>) => applyCoupon(coupons[parseInt(e.target.value)])
  return (
    <div className="mt-6 bg-white p-4 rounded shadow">
      <h2 className="text-2xl font-semibold mb-2">쿠폰 적용</h2>
      <select
        onChange={applyCouponHandler}
        className="w-full p-2 border rounded mb-2"
      >
```

요렇게 하면 onChange에서 어떤일이 발생하는지 좀더 간결하게 파악 가능하다고 생각되는데 어떠신가요??

Reply:
음 저는 개인적으로 전자가 더 깔끔하다고 생각하는데,
후자처럼 따로 로직을 빼서 작성하는게 더 좋은 방법일거 같네요
참고하겠습니다:)

Reply:
@cherrybbang 정답은 없습니다.

제가 생각해도 한줄 정도의 코드라면 굳이 구분하는 것이 사람에 따라 가독성을 해친다고 생각 할 수도 있을 것 같아요 ㅎㅎ
제 스타일이 로직은 숨기고 가독성 좋은 함수명을 만들어 쓰는 것을 지향하는 것 같습니다.